src/features/island/fisherman/FishingChallenge.tsx

```diff
@@ -0,0 +1,339 @@
+    return 1000;
+  }
+
+  if (attempts > 15) {
+    return 1200;
+  }
+
+  if (attempts > 10) {
+    return 1500;
+  }
+
+  if (attempts > 7) {
+    return 2000;
+  }
+
+  if (attempts > 3) {
+    return 2500;
+  }
+
+  return 3000;
+}
+
+// Difficult to health
+const FISH_HEALTH: Record<number, number> = {
+  1: 10,
+  2: 13,
+  3: 16,
+  4: 20,
+  5: 25,
+};
+
+interface Props {
+  onCatch: () => void;
+  onMiss: () => void;
+  difficulty?: number;
+  fishName: FishName | MarineMarvelName;
+}
+
+export const FishingChallengeIntro: React.FC<{ onNext: () => void }> = ({
+  onNext,
+}) => {
+  return (
+    <>
+      <div className="p-2">
+        <p className="text-sm">A powerful catch awaits!</p>
+        <p className="text-sm">Use all your strength to reel it in.</p>
+        <p className="text-sm">Stop the green bar on the fish to succeed</p>
+        <p className="text-sm">Be quick - 3 wrong tries, and it escapes!</p>
+      </div>
+      <Button onClick={onNext}>Next</Button>
+    </>
+  );
+};
+
+export const FishingChallenge: React.FC<Props> = ({
+  onCatch,
+  onMiss,
+  difficulty = 1,
+  fishName,
+}) => {
+  const [showIntro, setShowIntro] = useState(true);
+  const [state, setState] = useState<"idle" | "playing">("idle");
+
+  const [tentacleAngle, setTentacleAngle] = useState(getRandomAngle(150, 0));
+  const [attempts, setAttempts] = useState<Attempt[]>([]);
+
+  const direction = useRef<"clockwise" | "anticlockwise">("clockwise");
+  const [castFrom, setCastFrom] = useState(0);
+  const [castTo, setCastTo] = useState(360);
+
+  const health = FISH_HEALTH[difficulty] ?? 10;
+
+  const changeDirection = (fromDegrees: number) => {
+    const to =
+      direction.current === "clockwise" ? fromDegrees - 360 : fromDegrees + 360;
+    setCastTo(to);
+    setCastFrom(fromDegrees);
+    direction.current =
+      direction.current === "anticlockwise" ? "clockwise" : "anticlockwise";
+  };
+
+  const greenBarProps = useSpring({
+    from: { transform: `rotate(${castFrom}deg)` },
+    to: { transform: `rotate(${castTo}deg)` },
+    onRest: () => {
+      if (misses >= 2) {
+        onMiss();
+      }
+
+      changeDirection(castTo);
+      setAttempts((prev) => [...prev, "miss"]);
+    },
+    config: {
+      duration: getSpeed(attempts.length),
+    },
+    pause: state === "idle",
+  });
+
+  const misses = attempts.filter((a) => a === "miss").length;
+  const hits = attempts.filter((a) => a === "hit").length;
+
+  const tentacleProps = useSpring({
+    from: { transform: `rotate(${tentacleAngle}deg)` },
+    to: { transform: `rotate(${tentacleAngle}deg)` },
+    config: {
+      duration: 5000,
+    },
+  });
+
+  const barHeight = 10;
+
+  const reel = () => {
+    const greenBarAngle = extractAngle(greenBarProps.transform.get());
+    const tentacleAngle = extractAngle(tentacleProps.transform.get());
+
+    const degreeDifference = Math.abs(
+      normaliseAngle(greenBarAngle) - normaliseAngle(tentacleAngle)
+    );
```

Comment:
```suggestion
    const degreeDifference = Math.abs(
      ((normaliseAngle(greenBarAngle) - normaliseAngle(tentacleAngle) + 180) %
        360) -
        180
    );
```

I think this may return failed results around zero degrees. For example if the fish is at `5` deg and the green bar is at `358` degrees, the difference will be `353`